x/gamm/genesis.go

```diff
@@ -0,0 +1,60 @@
+package gamm
+
+import (
+	"fmt"
+
+	codectypes "github.com/cosmos/cosmos-sdk/codec/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/osmosis-labs/osmosis/x/gamm/keeper"
+	"github.com/osmosis-labs/osmosis/x/gamm/types"
+)
+
+// InitGenesis initializes the capability module's state from a provided genesis
+// state.
+func InitGenesis(ctx sdk.Context, k keeper.Keeper, genState types.GenesisState, unpacker codectypes.AnyUnpacker) {
```

Comment:
Why do pass in an unpacker, instead of having a types.UnpackPools method, ala https://github.com/cosmos/cosmos-sdk/blob/master/x/auth/genesis.go#L16

Reply:
I think doing a second iteration here would be worth it just for ease of readability in the future

Reply:
I tried GetCachedValue solution but it didn't work - just spent a lot of time already and finally staying here